niceml/data/datasets/dfdataset.py

```diff
@@ -27,51 +38,148 @@ class RegDataInfo(DataInfo):
 
-    def __init__(  # pylint: disable=too-many-arguments
+    def __init__(  # ruff: noqa: PLR0913
         self,
         id_key: str,
         batch_size: int,
         set_name: str,
-        df_loader: DfLoader,
-        df_path: str,
+        data_location: Union[dict, LocationConfig],
+        df_path: str = ExperimentFilenames.SUBSET_NAME,
         shuffle: bool = False,
+        data_shuffler: Optional[DataShuffler] = None,
+        dataframe_filters: Optional[List[DataframeFilter]] = None,
+        feature_combiners: Optional[List[FeatureCombiner]] = None,
+        extra_key_list: Optional[List[str]] = None,
     ):
+        """
+        The __init__ function is called when the class is instantiated.
+        It sets up the instance of the class, and defines all of its attributes.
+        The __init__ function takes in a number of arguments, which are used to set up
+        these attributes.
+
+        Args:
+            id_key: str: Specify the column name of the id column in your dataframe
+            batch_size: int: Define the size of each batch
+            set_name: str: Identify the dataset
+            data_location: Union[dict, LocationConfig]: Specify the location of the data
+            df_path: str: Specify the file name of the dataframe
+            shuffle: bool: Shuffle the data
+            dataframe_filters: Optional[List[DataframeFilter]]: Optional list of dataframe filters
+                                to filter the data
+
+        """
         super().__init__()
+        self.data_shuffler = data_shuffler or DefaultDataShuffler()
+        self.dataframe_filters = dataframe_filters or []
+        self.df_path = df_path
+        self.data_location = data_location
         self.batch_size = batch_size
         self.set_name = set_name
         self.id_key = id_key
-        self.dataframe: pd.DataFrame = df_loader.load_df(df_path)
-        self.index_list = list(self.dataframe.index)
+        self.index_list = []
         self.shuffle = shuffle
+        self.data: Optional[pd.DataFrame] = None
         self.inputs: List[dict] = []
         self.targets: List[dict] = []
-
-    def get_batch_size(self) -> int:
-        return self.batch_size
-
-    def get_set_name(self) -> str:
-        return self.set_name
+        self.extra_key_list: List[str] = extra_key_list or []
+        self.feature_combiners: List[FeatureCombiner] = feature_combiners or []
 
     def initialize(
         self, data_description: RegDataDescription, exp_context: ExperimentContext
     ):
+        """
+        The initialize function is called when the data set is created.
+        It takes in a `RegDataDescription` object, which contains information about the
+        inputs and targets of your dataset. The initialize function should also take in an
+        `ExperimentContext` object, which contains information about where to find your
+        data on disk. The `ExperimentContext` is not used in this class.
```

Comment:
I am sure the 'ExperimentContext' is of some use for the class. Please delete the last sentence about not using it.